core/src/mender-client.c

```diff
@@ -784,6 +789,19 @@ mender_client_initialization_work_function(void) {
             goto REBOOT;
         }
         free(storage_deployment_data);
+
+        /* Check deployment data version compatibility */
+        int          version = -1;
+        const cJSON *json_number;
+        if (NULL != (json_number = cJSON_GetObjectItemCaseSensitive(mender_client_deployment_data, "version"))) {
+            version = (int)cJSON_GetNumberValue(json_number);
```

Comment:
Worth a comment that the return type here is `double`. Another "amazing" fact about cJSON... :roll_eyes: